src/components/expanded-state/custom-gas/FeesPanel.js

```diff
@@ -524,6 +584,123 @@ export default function FeesPanel({
+      Alert({
+        buttons: [
+          {
+            onPress: () => onAlertProceeded(callback),
+            text: 'Proceed Anyway',
+          },
+          {
+            onPress: () => openCustomOptions(FOCUS_TO_MAX_BASE_FEE),
+            style: 'cancel',
+            text: 'Edit Max Base Fee',
+          },
+        ],
+        message: highAlert
+          ? ALERT_MESSAGE_LOWER
+          : ALERT_MESSAGE_HIGHER_MAX_BASE_FEE_NEEDED,
+        title: highAlert
+          ? ALERT_TITLE_LOWER_MAX_BASE_FEE_NEEDED
+          : ALERT_TITLE_HIGHER_MAX_BASE_FEE_NEEDED,
+      });
+    },
+    [maxBaseFeeWarning, onAlertProceeded, openCustomOptions]
+  );
+
+  const alertMaxPriority = useCallback(
+    callback => {
+      const highAlert = maxPriorityFeeWarning === HIGHER_THAN_NECESSARY;
+      Alert({
+        buttons: [
+          {
+            onPress: () => onAlertProceeded(callback),
+            text: 'Proceed Anyway',
+          },
+          {
+            onPress: () => openCustomOptions(FOCUS_TO_MINER_TIP),
+            style: 'cancel',
+            text: 'Edit Miner Tip',
+          },
+        ],
+        message: highAlert
+          ? ALERT_MESSAGE_LOWER
+          : ALERT_MESSAGE_HIGHER_MINER_TIP_NEEDED,
+        title: highAlert
+          ? ALERT_TITLE_LOWER_MINER_TIP_NEEDED
+          : ALERT_TITLE_HIGHER_MINER_TIP_NEEDED,
+      });
+    },
+    [maxPriorityFeeWarning, onAlertProceeded, openCustomOptions]
+  );
+
+  validateGasParams.current = callback => validateParams(callback);
+
+  const validateParams = useCallback(
+    callback => {
+      if (userProceededOnWarnings || !selectedOptionIsCustom) return;
+      const maxBaseValidated = !maxBaseFeeError && !maxBaseFeeWarning;
+      const maxPriorityValidated =
+        !maxPriorityFeeError && !maxPriorityFeeWarning;
+      if (!maxBaseValidated) {
+        alertMaxBaseFee(callback);
+      } else if (!maxPriorityValidated) {
+        alertMaxPriority(callback);
+      }
+    },
+    [
+      alertMaxBaseFee,
+      alertMaxPriority,
+      maxBaseFeeError,
+      maxBaseFeeWarning,
+      maxPriorityFeeError,
+      maxPriorityFeeWarning,
+      selectedOptionIsCustom,
+      userProceededOnWarnings,
+    ]
+  );
+
+  useEffect(() => {
+    const maxBaseValidated = !maxBaseFeeError && !maxBaseFeeWarning;
+    const maxPriorityValidated = !maxPriorityFeeError && !maxPriorityFeeWarning;
+    if (
+      !userProceededOnWarnings &&
+      selectedOptionIsCustom &&
+      (!maxBaseValidated || !maxPriorityValidated)
+    ) {
+      setCanGoBack(false);
+    } else {
+      setCanGoBack(true);
+    }
+  }, [
+    alertMaxBaseFee,
+    alertMaxPriority,
+    maxBaseFeeError,
+    maxBaseFeeWarning,
+    maxPriorityFeeError,
+    maxPriorityFeeWarning,
+    selectedOptionIsCustom,
+    setCanGoBack,
+    userProceededOnWarnings,
+  ]);
+
+  useEffect(() => {
+    return function validate() {
+      validateGasParams.current?.();
```

Comment:
why wrap this in a function vs calling it directly/how is this called, and where are we returning too 🤔

Reply:
this is to be able to run the validations on `unmount`, that way only when the component is being unmounted `function validate()` is going to be triggered

Reply:
yup just did some reading on this, thanks for the explanation 👌